src/main/java/uk/gov/cslearning/catalogue/repository/CourseRepositoryImpl.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.cslearning.catalogue.repository;
+
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.query.NativeSearchQueryBuilder;
+import org.springframework.data.elasticsearch.core.query.SearchQuery;
+import org.springframework.stereotype.Repository;
+import uk.gov.cslearning.catalogue.domain.Course;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+
+@Repository
+public class CourseRepositoryImpl {
+    private ElasticsearchOperations operations;
+
+    public CourseRepositoryImpl(ElasticsearchOperations operations) {
+        checkArgument(operations != null);
+        this.operations = operations;
+    }
+
+    public List<Course> findPublishedAndArchivedMandatoryCourses() {
+        
+        BoolQueryBuilder boolQuery = boolQuery();
+        boolQuery.should(QueryBuilders.matchQuery("status", "Published"));
```

Comment:
Worth creating a string constants?

Reply:
Again, these status are not reused in the system as of now but, these can be defined in a constant file.